nexus/src/nexus.rs

```diff
@@ -2249,6 +2259,49 @@ impl Nexus {
     pub async fn session_hard_delete(&self, token: String) -> DeleteResult {
         self.db_datastore.session_hard_delete(token).await
     }
+
+    pub async fn updates_refresh_metadata(&self) -> Result<(), Error> {
+        let rack = self.as_rack();
+        let trust_root = self
+            .tuf_trusted_root
+            .as_ref()
+            .ok_or_else(|| Error::InvalidRequest {
+                message: "updates system not configured".into(),
+            })?
+            .clone();
+
+        let artifacts = tokio::task::spawn_blocking(move || {
+            crate::updates::read_artifacts(&rack, &trust_root)
+        })
+        .await
+        // first, the JoinError:
+        .map_err(|e| Error::InternalError { internal_message: e.to_string() })?
+        // next, the boxed dyn Error:
+        .map_err(|e| Error::InternalError {
+            internal_message: e.to_string(),
+        })?;
+
+        // FIXME: if we hit an error in any of these database calls, the available artifact table
+        // will be out of sync with the current artifacts.json. can we do a transaction or
+        // something?
+
+        let mut current_version = None;
+        for artifact in artifacts {
+            current_version = Some(artifact.targets_role_version);
+            self.db_datastore
+                .update_available_artifact_upsert(artifact)
+                .await?;
+        }
+
+        // ensure table is in sync with current copy of artifacts.json
+        if let Some(current_version) = current_version {
+            self.db_datastore
+                .update_available_artifact_hard_delete_outdated(current_version)
+                .await?;
+        }
+
```

Comment:
When do you think we'd actually try to apply updates (basically, making the [calls to POST](https://github.com/oxidecomputer/omicron/pull/457/files#diff-a9cd239d5f95dd46c3f23f59130f3ed82f31aa3bc23434eeb569b50ecbde1d46R97) `/update` on each sled) to the rest of the rack? Would that be at the end of this function, or some other time?

Reply:
For the demo, probably immediately just for simplicity's sake?

At release I think a background task that periodically checks for out-of-date artifacts on the rack and starts applying updates is probably correct. Otherwise we need to handle applying updates here as well as when a sled-agent comes online (e.g. brand new sled, or sled that's been powered off for a few weeks).